virtcontainers/sandbox.go

```diff
@@ -781,12 +799,18 @@ func createSandbox(ctx context.Context, sandboxConfig SandboxConfig, factory Fac
 	}
 
 	// Below code path is called only during create, because of earlier check.
-	if err := s.agent.createSandbox(s); err != nil {
+	defer func() {
+		if err != nil {
+			rollbackFailingSandboxCreation(ctx, s)
```

Comment:
You should not call this global function from here. It makes sense from `virtcontainers/api.go`, but here instead, we should have a finer granularity, by deferring a bunch of rollbacks after each function call succeeds. Something along those lines:
```go
if err = s.agent.createSandbox(s); err != nil {
	return nil, err
}

defer func() {
	if err != nil {
                s.agent.cleanupSandbox(s)
        }
}
```

Reply:
updated~